tensornetwork/network_components.py

```diff
@@ -232,6 +237,82 @@ def __str__(self) -> Text:
     return self.name
 
 
+class CopyNode(Node):
+  def __init__(self,
+               rank: int,
+               dimension: int,
+               name: Text,
+               axis_names: List[Text],
+               backend: base_backend.BaseBackend) -> None:
+    # TODO: Make this computation lazy, once Node doesn't require tensor
+    # at instatiation.
+    copy_tensor = self.make_copy_tensor(rank, dimension)
+    copy_tensor = backend.convert_to_tensor(copy_tensor)
+    super().__init__(copy_tensor, name, axis_names, backend)
+
+  @staticmethod
+  def make_copy_tensor(rank: int, dimension: int) -> Tensor:
+    shape = (dimension,) * rank
+    copy_tensor = np.zeros(shape)
```

Comment:
We'll have to deal with the `dtype` as well. Not sure whether to do that at the `TensorNetwork` or `make_copy_tensor` level.

Reply:
Done.